diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index b814b33..21dc6ca 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -187,7 +187,7 @@ include(jxl.cmake)
 if(JPEGXL_ENABLE_TOOLS OR BUILD_TESTING)
   include(jxl_extras.cmake)
 endif()
-include(jxl_threads.cmake)
+#include(jxl_threads.cmake)
 
 # For simplicity all the library headers, both source and generated ones, are
 # gathered in the binary folder. There is no distinction on which libraries use
diff --git a/lib/jxl.cmake b/lib/jxl.cmake
index 8d962e6..876509a 100644
--- a/lib/jxl.cmake
+++ b/lib/jxl.cmake
@@ -219,6 +219,7 @@ target_link_libraries(jxl PUBLIC ${JPEGXL_COVERAGE_FLAGS} jxl_base)
 target_link_libraries(jxl PUBLIC jxl_cms)
 target_link_libraries(jxl PRIVATE ${JPEGXL_INTERNAL_SHARED_LIBS})
 set_target_properties(jxl PROPERTIES
+  NO_SONAME 1
   VERSION ${JPEGXL_LIBRARY_VERSION}
   SOVERSION ${JPEGXL_LIBRARY_SOVERSION})
 
diff --git a/lib/jxl_cms.cmake b/lib/jxl_cms.cmake
index 582b842..b80d36d 100644
--- a/lib/jxl_cms.cmake
+++ b/lib/jxl_cms.cmake
@@ -9,6 +9,7 @@ include(jxl_lists.cmake)
 include(GenerateExportHeader)
 
 add_library(jxl_cms
+  STATIC
   ${JPEGXL_INTERNAL_CMS_SOURCES}
 )
 target_compile_options(jxl_cms PRIVATE "${JPEGXL_INTERNAL_FLAGS}")
