diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf2fa1e..75f5773 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -588,7 +588,7 @@ if(WITH_JAVA)
   add_subdirectory(java)
 endif()
 
-if(ENABLE_SHARED)
+if(ENABLE_SHARED AND WITH_TURBOJPEG)
   add_subdirectory(sharedlib)
 endif()
 
@@ -679,6 +679,16 @@ if(WITH_TURBOJPEG)
   endif()
 endif()
 
+set(PSJPEG_SOURCES ${JPEG_SOURCES} $<TARGET_OBJECTS:simd> ${SIMD_OBJS} psjpeg.c)
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+add_library(psjpeg SHARED ${PSJPEG_SOURCES})
+set_target_properties(psjpeg PROPERTIES DEFINE_SYMBOL DLLDEFINE)
+
+if(UNIX AND HAVE_VERSION_SCRIPT)
+  set_target_properties(psjpeg PROPERTIES LINK_FLAGS
+    "-Wl,--version-script='${CMAKE_CURRENT_SOURCE_DIR}/psjpeg.ver'")
+endif()
+
 if(WIN32)
   set(USE_SETMODE "-DUSE_SETMODE")
 endif()
@@ -1475,6 +1485,19 @@ if(WITH_TURBOJPEG)
   endif()
 endif()
 
+install(TARGETS psjpeg EXPORT ${CMAKE_PROJECT_NAME}Targets
+  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+if(MSVC AND CMAKE_C_LINKER_SUPPORTS_PDB)
+  install(FILES "$<TARGET_PDB_FILE:psjpeg>"
+    DESTINATION ${CMAKE_INSTALL_BINDIR} OPTIONAL)
+endif()
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/psjpeg.h ${CMAKE_CURRENT_BINARY_DIR}/jconfig.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/jerror.h ${CMAKE_CURRENT_SOURCE_DIR}/jmorecfg.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/jpeglib.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/psjpeg)
+
 if(ENABLE_STATIC)
   install(TARGETS jpeg-static EXPORT ${CMAKE_PROJECT_NAME}Targets
     INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
@@ -1519,9 +1542,9 @@ if((UNIX OR MINGW) AND INSTALL_DOCS)
     ${CMAKE_CURRENT_SOURCE_DIR}/wrjpgcom.1
     DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
 endif()
+if(WITH_TURBOJPEG)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/pkgscripts/libjpeg.pc
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
-if(WITH_TURBOJPEG)
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/pkgscripts/libturbojpeg.pc
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
