diff --git a/src/jdapistd.c b/src/jdapistd.c
index 8f5f85a..6b66964 100644
--- a/src/jdapistd.c
+++ b/src/jdapistd.c
@@ -203,8 +203,8 @@ _jpeg_crop_scanline(j_decompress_ptr cinfo, JDIMENSION *xoffset,
   if (cinfo->master->lossless || cinfo->raw_data_out)
     ERREXIT(cinfo, JERR_NOTIMPL);
 
-  if ((cinfo->global_state != DSTATE_SCANNING &&
-       cinfo->global_state != DSTATE_BUFIMAGE) || cinfo->output_scanline != 0)
+  if ((cinfo->global_state < DSTATE_SCANNING ||
+       cinfo->global_state > DSTATE_BUFIMAGE) || cinfo->output_scanline != 0)
     ERREXIT1(cinfo, JERR_BAD_STATE, cinfo->global_state);
 
   if (!xoffset || !width)
diff --git a/src/jddctmgr.c b/src/jddctmgr.c
index 60bae88..26d3f55 100644
--- a/src/jddctmgr.c
+++ b/src/jddctmgr.c
@@ -125,10 +125,14 @@ start_pass(j_decompress_ptr cinfo)
         method_ptr = _jpeg_idct_2x2;
       method = JDCT_ISLOW;      /* jidctred uses islow-style table */
       break;
+#endif
+#ifdef IDCT_SCALING_SUPPORTED_ODD
     case 3:
       method_ptr = _jpeg_idct_3x3;
       method = JDCT_ISLOW;      /* jidctint uses islow-style table */
       break;
+#endif
+#ifdef IDCT_SCALING_SUPPORTED
     case 4:
 #ifdef WITH_SIMD
       if (jsimd_set_idct_4x4(cinfo))
@@ -138,6 +142,8 @@ start_pass(j_decompress_ptr cinfo)
         method_ptr = _jpeg_idct_4x4;
       method = JDCT_ISLOW;      /* jidctred uses islow-style table */
       break;
+#endif
+#ifdef IDCT_SCALING_SUPPORTED_ODD
     case 5:
       method_ptr = _jpeg_idct_5x5;
       method = JDCT_ISLOW;      /* jidctint uses islow-style table */
@@ -191,7 +197,7 @@ start_pass(j_decompress_ptr cinfo)
         break;
       }
       break;
-#ifdef IDCT_SCALING_SUPPORTED
+#ifdef IDCT_SCALING_SUPPORTED_ODD
     case 9:
       method_ptr = _jpeg_idct_9x9;
       method = JDCT_ISLOW;      /* jidctint uses islow-style table */
diff --git a/src/jdmaster.c b/src/jdmaster.c
index ac6183b..260b607 100644
--- a/src/jdmaster.c
+++ b/src/jdmaster.c
@@ -123,6 +123,7 @@ jpeg_core_output_dimensions(j_decompress_ptr cinfo)
         jdiv_round_up((long)cinfo->image_height * 2L, (long)DCTSIZE);
       cinfo->_min_DCT_h_scaled_size = 2;
       cinfo->_min_DCT_v_scaled_size = 2;
+#ifdef IDCT_SCALING_SUPPORTED_ODD
     } else if (cinfo->scale_num * DCTSIZE <= cinfo->scale_denom * 3) {
       /* Provide 3/block_size scaling */
       cinfo->output_width = (JDIMENSION)
@@ -131,6 +132,7 @@ jpeg_core_output_dimensions(j_decompress_ptr cinfo)
         jdiv_round_up((long)cinfo->image_height * 3L, (long)DCTSIZE);
       cinfo->_min_DCT_h_scaled_size = 3;
       cinfo->_min_DCT_v_scaled_size = 3;
+#endif
     } else if (cinfo->scale_num * DCTSIZE <= cinfo->scale_denom * 4) {
       /* Provide 4/block_size scaling */
       cinfo->output_width = (JDIMENSION)
@@ -139,6 +141,7 @@ jpeg_core_output_dimensions(j_decompress_ptr cinfo)
         jdiv_round_up((long)cinfo->image_height * 4L, (long)DCTSIZE);
       cinfo->_min_DCT_h_scaled_size = 4;
       cinfo->_min_DCT_v_scaled_size = 4;
+#ifdef IDCT_SCALING_SUPPORTED_ODD
     } else if (cinfo->scale_num * DCTSIZE <= cinfo->scale_denom * 5) {
       /* Provide 5/block_size scaling */
       cinfo->output_width = (JDIMENSION)
@@ -164,6 +167,9 @@ jpeg_core_output_dimensions(j_decompress_ptr cinfo)
       cinfo->_min_DCT_h_scaled_size = 7;
       cinfo->_min_DCT_v_scaled_size = 7;
     } else if (cinfo->scale_num * DCTSIZE <= cinfo->scale_denom * 8) {
+#else
+    } else {
+#endif
       /* Provide 8/block_size scaling */
       cinfo->output_width = (JDIMENSION)
         jdiv_round_up((long)cinfo->image_width * 8L, (long)DCTSIZE);
@@ -171,6 +177,7 @@ jpeg_core_output_dimensions(j_decompress_ptr cinfo)
         jdiv_round_up((long)cinfo->image_height * 8L, (long)DCTSIZE);
       cinfo->_min_DCT_h_scaled_size = 8;
       cinfo->_min_DCT_v_scaled_size = 8;
+#ifdef IDCT_SCALING_SUPPORTED_ODD
     } else if (cinfo->scale_num * DCTSIZE <= cinfo->scale_denom * 9) {
       /* Provide 9/block_size scaling */
       cinfo->output_width = (JDIMENSION)
@@ -235,6 +242,7 @@ jpeg_core_output_dimensions(j_decompress_ptr cinfo)
         jdiv_round_up((long)cinfo->image_height * 16L, (long)DCTSIZE);
       cinfo->_min_DCT_h_scaled_size = 16;
       cinfo->_min_DCT_v_scaled_size = 16;
+#endif
     }
 
     /* Recompute dimensions of components */
diff --git a/src/jidctint.c b/src/jidctint.c
index 81b71db..c3019cc 100644
--- a/src/jidctint.c
+++ b/src/jidctint.c
@@ -435,7 +435,7 @@ _jpeg_idct_islow(j_decompress_ptr cinfo, jpeg_component_info *compptr,
   }
 }
 
-#ifdef IDCT_SCALING_SUPPORTED
+#ifdef IDCT_SCALING_SUPPORTED_ODD
 
 
 /*
diff --git a/src/jmorecfg.h b/src/jmorecfg.h
index a4df71c..47eb581 100644
--- a/src/jmorecfg.h
+++ b/src/jmorecfg.h
@@ -27,7 +27,7 @@
  * bytes of storage, whether actually used in an image or not.)
  */
 
-#define MAX_COMPONENTS  10      /* maximum number of image components */
+#define MAX_COMPONENTS   4      /* maximum number of image components */
 
 
 /*
@@ -236,8 +236,6 @@ typedef int boolean;
 /* Capability options common to encoder and decoder: */
 
 #define DCT_ISLOW_SUPPORTED     /* accurate integer method */
-#define DCT_IFAST_SUPPORTED     /* less accurate int method [legacy feature] */
-#define DCT_FLOAT_SUPPORTED     /* floating-point method [legacy feature] */
 
 /* Encoder capability options: */
 
@@ -245,7 +243,6 @@ typedef int boolean;
 #define C_PROGRESSIVE_SUPPORTED     /* Progressive JPEG?  (Requires
                                        C_MULTISCAN_FILES_SUPPORTED and
                                        ENTROPY_OPT_SUPPORTED) */
-#define C_LOSSLESS_SUPPORTED        /* Lossless JPEG? */
 #define ENTROPY_OPT_SUPPORTED       /* Optimization of entropy coding parms? */
 /* Note: if you selected 12-bit data precision, it is dangerous to turn off
  * ENTROPY_OPT_SUPPORTED.  The standard Huffman tables are only good for 8-bit
@@ -255,22 +252,15 @@ typedef int boolean;
  * The exact same statements apply for lossless JPEG: the default tables don't
  * work for lossless mode.  (This may get fixed, however.)
  */
-#define INPUT_SMOOTHING_SUPPORTED   /* Input image smoothing option? */
 
 /* Decoder capability options: */
 
 #define D_MULTISCAN_FILES_SUPPORTED /* Multiple-scan JPEG files? */
 #define D_PROGRESSIVE_SUPPORTED     /* Progressive JPEG?  (Requires
                                        D_MULTISCAN_FILES_SUPPORTED) */
-#define D_LOSSLESS_SUPPORTED        /* Lossless JPEG?  (Requires
-                                       D_MULTISCAN_FILES_SUPPORTED) */
 #define SAVE_MARKERS_SUPPORTED      /* jpeg_save_markers() needed? */
-#define BLOCK_SMOOTHING_SUPPORTED   /* Block smoothing? (Progressive only) */
 #define IDCT_SCALING_SUPPORTED      /* Output rescaling via IDCT?  (Requires
                                        DCT_ISLOW_SUPPORTED) */
-#define UPSAMPLE_MERGING_SUPPORTED  /* Fast path for sloppy upsampling? */
-#define QUANT_1PASS_SUPPORTED       /* 1-pass color quantization? */
-#define QUANT_2PASS_SUPPORTED       /* 2-pass color quantization? */
 
 /* more capability options later, no doubt */
 
diff --git a/src/jconfig.h.in b/src/jconfig.h.in
index 6cb8296..046dac5 100644
--- a/src/jconfig.h.in
+++ b/src/jconfig.h.in
@@ -38,11 +38,8 @@
 
 #undef RIGHT_SHIFT_IS_UNSIGNED
 
-/* Define "boolean" as unsigned char, not int, per Windows custom */
-#ifndef __RPCNDR_H__            /* don't conflict if rpcndr.h already read */
-typedef unsigned char boolean;
-#endif
-#define HAVE_BOOLEAN            /* prevent jmorecfg.h from redefining it */
+#define NO_GETENV
+#define NO_PUTENV
 
 /* Define "INT32" as int, not long, per Windows custom */
 #if !(defined(_BASETSD_H_) || defined(_BASETSD_H))   /* don't conflict if basetsd.h already read */
