diff --git a/pngread.c b/pngread.c
index d82ffbd..f0de0f8 100644
--- a/pngread.c
+++ b/pngread.c
@@ -222,7 +222,7 @@ png_read_frame_head(png_structp png_ptr, png_infop info_ptr)
          png_ensure_sequence_number(png_ptr, length);
 
          /* Discard trailing fdATs for all frames except the first. */
-         if (!have_chunk_after_DAT && png_ptr->num_frames_read > 1)
+         if (!have_chunk_after_DAT && png_ptr->num_frames_read >= 1)
          {
             png_crc_finish(png_ptr, length - 4);
          }
@@ -526,7 +526,7 @@ png_read_row(png_structrp png_ptr, png_bytep row, png_bytep dsp_row)
    png_ptr->row_buf[0]=255; /* to force error if no data was found */
    png_read_IDAT_data(png_ptr, png_ptr->row_buf, row_info.rowbytes + 1);
 
-   if (png_ptr->row_buf[0] > PNG_FILTER_VALUE_NONE)
+   if (png_ptr->row_buf[0] > PNG_FILTER_VALUE_NONE && png_ptr->row_buf[0] < PNG_FILTER_VALUE_LAST)
    {
       if (png_ptr->row_buf[0] < PNG_FILTER_VALUE_LAST)
          png_read_filter_row(png_ptr, &row_info, png_ptr->row_buf + 1,
diff --git a/pngwrite.c b/pngwrite.c
index ee88e26..8faee9e 100644
--- a/pngwrite.c
+++ b/pngwrite.c
@@ -1548,12 +1548,6 @@ png_write_frame_head(png_structp png_ptr, png_infop info_ptr,
 {
    png_debug(1, "in png_write_frame_head");
 
-   /* There is a chance this has been set after png_write_info was called,
-    * so it would be set but not written. Is there a way to be sure?
-    */
-   if (!(info_ptr->valid & PNG_INFO_acTL))
-      png_error(png_ptr, "Cannot write APNG frame: missing acTL");
-
    png_write_reset(png_ptr);
 
    png_write_reinit(png_ptr, info_ptr, width, height);
diff --git a/pngwutil.c b/pngwutil.c
index ad1baca..a1fe448 100644
--- a/pngwutil.c
+++ b/pngwutil.c
@@ -2900,11 +2900,6 @@ png_write_reinit(png_structp png_ptr, png_infop info_ptr,
        height > png_ptr->first_frame_height)
       png_error(png_ptr, "Oversized frame in fcTL");
 
-   png_set_IHDR(png_ptr, info_ptr, width, height,
-                info_ptr->bit_depth, info_ptr->color_type,
-                info_ptr->interlace_type, info_ptr->compression_type,
-                info_ptr->filter_type);
-
    png_ptr->width = width;
    png_ptr->height = height;
    png_ptr->rowbytes = PNG_ROWBYTES(png_ptr->pixel_depth, width);
