diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6904296..f154b1c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -752,6 +752,20 @@ if(PNG_FRAMEWORK)
                         PUBLIC ${PNG_LINK_LIBRARIES})
 endif()
 
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(PNG_LIBRARY_TARGETS pspng)
+set(pspng_sources ${libpng_sources} pspng.h pspng.c)
+add_library(pspng SHARED ${pspng_sources})
+set_target_properties(pspng PROPERTIES DEFINE_SYMBOL DLLDEFINE)
+add_dependencies(pspng png_genfiles)
+target_include_directories(pspng PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
+target_link_libraries(pspng PUBLIC ${PNG_LINK_LIBRARIES})
+
+if(UNIX AND HAVE_LD_VERSION_SCRIPT)
+  set_target_properties(pspng PROPERTIES LINK_FLAGS
+    "-Wl,--version-script='${CMAKE_CURRENT_SOURCE_DIR}/pspng.ver'")
+endif()
+
 if(NOT PNG_LIBRARY_TARGETS)
   message(SEND_ERROR "No library variant selected to build. "
                      "Please enable at least one of the following options: "
@@ -1170,6 +1184,15 @@ if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL)
   endif()
 endif()
 
+install(TARGETS ${PNG_LIBRARY_TARGETS}
+        EXPORT pspng
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install(FILES ${libpng_public_hdrs} pspng.h
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/pspng COMPONENT include)
+
 # Create an export file that CMake users can include() to import our targets.
 if(NOT SKIP_INSTALL_EXPORT AND NOT SKIP_INSTALL_ALL)
   install(EXPORT libpng
