diff --git a/pngread.c b/pngread.c
index fa44d5a..4736f92 100644
--- a/pngread.c
+++ b/pngread.c
@@ -320,7 +320,7 @@ png_read_frame_head(png_structp png_ptr, png_infop info_ptr)
             png_ensure_sequence_number(png_ptr, length);
 
             /* discard trailing fdATs for frames other than the first */
-            if (!have_chunk_after_DAT && png_ptr->num_frames_read > 1)
+            if (!have_chunk_after_DAT && png_ptr->num_frames_read >= 1)
                 png_crc_finish(png_ptr, length - 4);
             else if(png_ptr->mode & PNG_HAVE_fcTL)
             {
diff --git a/pngwrite.c b/pngwrite.c
index 5e68032..cf99bf9 100644
--- a/pngwrite.c
+++ b/pngwrite.c
@@ -1480,11 +1480,6 @@ png_write_frame_head(png_structp png_ptr, png_infop info_ptr,
 {
     png_debug(1, "in png_write_frame_head");
 
-    /* there is a chance this has been set after png_write_info was called,
-    * so it would be set but not written. is there a way to be sure? */
-    if (!(info_ptr->valid & PNG_INFO_acTL))
-        png_error(png_ptr, "png_write_frame_head(): acTL not set");
-
     png_write_reset(png_ptr);
 
     png_write_reinit(png_ptr, info_ptr, width, height);
diff --git a/pngwutil.c b/pngwutil.c
index b4809cd..6c66eb1 100644
--- a/pngwutil.c
+++ b/pngwutil.c
@@ -2902,11 +2902,6 @@ png_write_reinit(png_structp png_ptr, png_infop info_ptr,
         png_error(png_ptr, "width and/or height for a frame greater than"
                            "the ones in IHDR");
 
-    png_set_IHDR(png_ptr, info_ptr, width, height,
-                 info_ptr->bit_depth, info_ptr->color_type,
-                 info_ptr->interlace_type, info_ptr->compression_type,
-                 info_ptr->filter_type);
-
     png_ptr->width = width;
     png_ptr->height = height;
     png_ptr->rowbytes = PNG_ROWBYTES(png_ptr->pixel_depth, width);
