diff --git a/libheif/plugins/decoder_dav1d.cc b/libheif/plugins/decoder_dav1d.cc
--- a/libheif/plugins/decoder_dav1d.cc
+++ b/libheif/plugins/decoder_dav1d.cc
@@ -106,5 +106,7 @@ heif_error dav1d_new_decoder2(void** dec, const heif_decoder_plugin_options* options)
   decoder->settings.all_layers = 0;
 
+  decoder->settings.n_threads = 1;
+
   if (options->num_threads) {
     decoder->settings.n_threads = options->num_threads;
   }
