diff --git a/CMakeLists.txt b/CMakeLists.txt
index e37e0121c..28fa981b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -141,6 +141,7 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libheif.pc
 # ---
 
 option(WITH_EXAMPLES "Build examples" ON)
+option(WITH_GDK_PIXBUF "Integrate gdk-pixbuf" OFF)
 
 option(WITH_DEFLATE_HEADER_COMPRESSION OFF)
 option(ENABLE_MULTITHREADING_SUPPORT "Switch off for platforms without multithreading support" ON)
@@ -150,8 +151,10 @@ if(WITH_EXAMPLES)
     add_subdirectory (examples)
 endif()
 add_subdirectory (libheif)
-add_subdirectory (gdk-pixbuf)
-add_subdirectory (gnome)
+if(WITH_GDK_PIXBUFF)
+    add_subdirectory (gdk-pixbuf)
+    add_subdirectory (gnome)
+endif()
 
 
 # --- packaging (source code)
